src/common/atomic_queue.cpp

```diff
@@ -559,6 +689,355 @@ void *process_queue(void *arg) {
 
             break;
         }
+
+        case ATOMIC_SCATTER_INDEX:
+            // Atomic scatter - indexed
+            if (msgPointer->flag & ATOMIC_WRITE_COMPLETED) {
+                // last write completed. If buffer is still allocated,
+                // deallocate and  remove the entry
+                if (msgPointer->flag & ATOMIC_BUFFER_ALLOCATED) {
+                    try {
+                        allocator->deallocate(ATOMIC_REGION_ID,
```

Comment:
Can this deallocate error be ignored? We might have completed deallocated, But ATOMIC_BUFFER_ALLOCATED might be still set.